### PR TITLE
Graphics

### PR DESCRIPTION
**Pull request by secretrobotron**
Merged new upstream changes from Alan and added a whole bunch more code to get us much closer to actually rendering things.

### CODE REVIEW

File: src/graphics/component/model.js [hunk 1]
```diff
@@ -7,18 +7,25 @@ define( function ( require ) {
     return function( engine ) {        
         
         var math = engine.math;
-        var Component = require( '../component' );
-        var Event = require( '../event' );
+        var Component = require( '../../core/component' );
```
Comment (modeswitch) [inline src/graphics/component/model.js:10]: Explain?

Comment (secretrobotron) [inline src/graphics/component/model.js:10]: require couldn't find these modules unless I added "../core". Found it while trying to make tests pass.

Comment (modeswitch) [inline src/graphics/component/model.js:10]: Oh, heh. The original path was wrong. OK, makes sense now.

File: src/graphics/service.js [hunk 2]
```diff
@@ -11,20 +22,71 @@ define( function ( require ) {
...
         var Graphics = function( options ) {
-            
+
+            var _scenes = [],
+                _renderTask,
+                _renderedFrames = 0,
+                _this = this;
+
+            engine.sceneAdded.subscribe( function( scene ) {
+                _scenes.push( scene );
```
Comment (modeswitch) [inline src/graphics/service.js:43]: This will prevent unused scenes from being garbage collected.

**Pull request merged**